feature/groups/src/main/java/com/mifos/feature/groupsList/presentation/GroupsListScreen.kt

```diff
@@ -0,0 +1,383 @@
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(it)
+                    .padding(PaddingSmall)
+            ) {
+                Crossfade(
+                    targetState = uiState,
+                    label = "GroupsList::State",
+                ) { state ->
+                    when (state) {
+                        is GroupsListState.Empty -> {}
+
+                        is GroupsListState.Error -> {
+                            MifosSweetError(
+                                message = state.message,
+                                onclick = onRefresh,
+                            )
+                        }
+
+                        is GroupsListState.GroupsFromAPI -> {
+                            GroupsListItemFromAPI(
+                                lazyListState = lazyListState,
+                                data = state.groups.collectAsLazyPagingItems(),
+                                doesSelected = {
+                                    selectedItems.contains(it)
+                                },
+                                doesInSelectionMode = {
+                                    selectedItems.isNotEmpty()
+                                },
+                                onGroupClick = onGroupClick,
+                                onSelectItem = {
+                                    if (selectedItems.contains(it)) {
+                                        selectedItems.remove(it)
+                                    }else {
+                                        selectedItems.add(it)
+                                    }
+                                },
+                                onRefresh = onRefresh
+                            )
+                        }
+
+                        is GroupsListState.GroupsFromLocalDB -> {
+                            GroupsListItemFromLocalDB(
+                                lazyListState = lazyListState,
+                                data = state,
+                                doesSelected = {
+                                    selectedItems.contains(it)
+                                },
+                                doesInSelectionMode = {
+                                    selectedItems.isNotEmpty()
+                                },
+                                onGroupClick = onGroupClick,
+                                onSelectItem = {
+                                    if (selectedItems.contains(it)) {
+                                        selectedItems.remove(it)
+                                    }else {
+                                        selectedItems.add(it)
+                                    }
+                                },
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun GroupsListItemFromAPI(
```

Comment:
There should only one GroupList screen that should be visible from one data source and it should not know from where data is coming local or server.

Reply:
Basically, it will check the user status in ViewModel if the user is online then will show data's from API otherwise will show data from Local DB